editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/adopters/AbstractProjectTemplate.java

```diff
@@ -47,18 +48,18 @@ public boolean supportsDSL(CamelDSLType type) {
 	 * @return	true on success, otherwise false
 	 * @throws CoreException	on any error
 	 */
-	public final boolean create(IProject project, NewProjectMetaData projectMetaData) throws CoreException {
-		IProgressMonitor monitor = new NullProgressMonitor();
+	public final boolean create(IProject project, NewProjectMetaData projectMetaData, IProgressMonitor monitor) throws CoreException {
+		SubMonitor subMonitor = SubMonitor.convert(monitor, 100);
```

Comment:
I'm pretty new to the SubMonitor, but seems like you use this differently in the rest of this PR.  The division of the work into arbitrary chunks (e.g., 30 vs. 15, etc.) seems just that, arbitrary.  Would it not be better to use the approach you take in other areas of the code and just give each individual task a "1"?  Maybe I'm missing something though...